owncloudApp/src/main/java/com/owncloud/android/ViewModelFactory.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.owncloud.android
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+
+class ViewModelFactory {
```

Comment:
If you had used Koin, you would have simplified all this logic 😄

Reply:
Koin or another dependency injector

Reply:
I tried it but when you have viewmodels and repositories with parameters Koin didn't look that straightforward... As you know, dependency injection is basically passing arguments and since we are not passing arguments from pretty distant parts, I decided to implement the dependency injection on my own, without adding more external dependencies.